### ISSUE TITLE
Create diagram showing modes in `AutoML`

### ISSUE THREAD

**Issue by pplonski** [mljar/docs#1]
There are 3 modes available:
- `Explain` - To be used when the user wants to explain and understand the data.
- `Perform` - To be used when the user wants to train a model that will be used in real-life use cases
- `Compete`- To be used for machine learning competitions (maximum performance)

The `Explain` mode:
- Uses algorithms: `Baseline`, `Linear`, `Decision Tree`, `Random Forest`, `XGBoost`, `Neural Network`, and `Ensemble`
- Uses 75%/25% train/test split
- Has full explanations in reports: learning curves, importance plots, and SHAP plots. (`explain_level=2`)
- Steps: `simple_algorithms`, `default_algorithms`,  `ensemble`

The `Perform` mode: 
- Uses the following models: `Linear`, `Random Forest`, `LightGBM`, `XGBoost`, `CatBoost`, `Neural Network`, and `Ensemble`.
- Uses a 5-fold CV (Cross-Validation) with shuffle and stratification.
- Has learning curves and importance plots in reports. (`explain_level=1`)
- Steps: `simple_algorithms`, `default_algorithms`, `not_so_random`, `golden_features`, `insert_random_feature`, `feature_selection`, `hill_climbing_1`, `hill_climbing_2`,  `ensemble`
- There are 4 models created for each algorithm in `not_so_random` step.
- There are 2 top models for each algorithm tuned in `hill_climbing` step.

The `Compete` mode: 
- Uses the following models: `Linear`, `DecisionTree`, `Random Forest`, `Extra Trees`, `XGBoost`, `CatBoost`, `Neural Network`, `Nearest Neighbors`, `Ensemble`, and `Stacking`.
- Uses 10-fold CV (Cross-Validation) with shuffle and stratification.
- It has only learning curves in the reports. (`explain_level=0`)
- Steps: `simple_algorithms`, `default_algorithms`, `not_so_random`, `golden_features`, `insert_random_feature`, `feature_selection`, `hill_climbing_1`, `hill_climbing_2`,  `ensemble`, `stack`, `ensemble_stacked`
- There are 9 models created for each algorithm in `not_so_random` step.
- There are 3 top models for each algorithm tuned in `hill_climbing` step.

**Comment by aplonska** [comment]
<table class="tg" style="undefined;table-layout: fixed; width: 687px">
<colgroup>
<col style="width: 155px">
<col style="width: 154px">
<col style="width: 203px">
<col style="width: 175px">
</colgroup>
<thead>
  <tr>
    <th class="tg-c4ne" colspan="4">Available modes</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-0pky"></td>
    <td class="tg-uypb">explain</td>
    <td class="tg-2xbj">perform</td>
    <td class="tg-uypb">compete</td>
  </tr>
  <tr>
    <td class="tg-c3ow" colspan="4">Uses models:</td>
  </tr>
  <tr>
    <td class="tg-0pky">Baseline</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47"></td>
  </tr>
  <tr>
    <td class="tg-0pky">Linear</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Decision Tree</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Random Forest</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">XGBoost</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Neural Network</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Ensemble</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Light GBM</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47"></td>
  </tr>
  <tr>
    <td class="tg-0pky">CatBoost</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Extra Trees</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Nearest Neighbors</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Stacking</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-c3ow" colspan="4">Validation:</td>
  </tr>
  <tr>
    <td class="tg-0pky"></td>
    <td class="tg-phtq">Uses 75%/25% train/test split</td>
    <td class="tg-0pky">Uses a 5-fold CV   (Cross-Validation) with shuffle and stratification</td>
    <td class="tg-phtq">Uses 10-fold CV   (Cross-Validation) with shuffle and stratification</td>
  </tr>
  <tr>
    <td class="tg-zqiz" colspan="4">Explanations:</td>
  </tr>
  <tr>
    <td class="tg-c3ow"></td>
    <td class="tg-svo0">explain_level=2</td>
    <td class="tg-c3ow">explain_level=1</td>
    <td class="tg-svo0">explain_level=0</td>
  </tr>
  <tr>
    <td class="tg-wp8o" colspan="4">Steps:</td>
  </tr>
  <tr>
    <td class="tg-0pky">Learning   curves </td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">Importance   plots </td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-7ykn"></td>
  </tr>
  <tr>
    <td class="tg-0pky">SHAP   plots</td>
    <td class="tg-zv47">+</td>
    <td class="tg-x1hj"></td>
    <td class="tg-7ykn"></td>
  </tr>
  <tr>
    <td class="tg-c3ow">Steps</td>
    <td class="tg-zv47"></td>
    <td class="tg-9d8n"></td>
    <td class="tg-zv47"></td>
  </tr>
  <tr>
    <td class="tg-0pky">simple_algorithms</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">default_algorithms</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">not_so_random</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">golden_features</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">insert_random_feature</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">feature_selection</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">hill_climbing_1</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">hill_climbing_2</td>
    <td class="tg-7ykn"></td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">ensemble</td>
    <td class="tg-zv47">+</td>
    <td class="tg-9d8n">+</td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">stack</td>
    <td class="tg-7ykn"></td>
    <td class="tg-x1hj"></td>
    <td class="tg-zv47">+</td>
  </tr>
  <tr>
    <td class="tg-0pky">ensemble_stacked</td>
    <td class="tg-7ykn"></td>
    <td class="tg-x1hj"></td>
    <td class="tg-zv47">+</td>
  </tr>
</tbody>
</table>

**Comment by aplonska** [comment]
Is it helpful?

**Comment by pplonski** [comment]
Thank you @aplonska ! That's what I need!
